presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -227,6 +237,28 @@ void PrestoServer::run() {
   registerShuffleInterfaceFactories();
   registerCustomOperators();
 
+  // These checks for connector factories can be removed after we remove the
+  // registrations from the Velox library.
+  if (!connector::hasConnectorFactory(
+          connector::hive::HiveConnectorFactory::kHiveConnectorName)) {
+    connector::registerConnectorFactory(
+        std::make_shared<connector::hive::HiveConnectorFactory>());
+    dwio::common::registerFileSinks();
```

Comment:
@majetideepak : Understand that these are linked with HiveConnector for now. But maybe filesinks, reader and writers are independent from the connectors as such. e.g. If we have just IcebergConnector and TpchConnector, we would still want the ParquetReader. 

So we should separate those registrations I feel.

wdyt ?

Reply:
@aditi-pandit I am just copying the Velox side registrations for now https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/HiveConnector.cpp#L135
There is work in terms of decoupling these modules.

> If we have just IcebergConnector and TpchConnector, we would still want the ParquetReader.

IcebergConnector still requires the HiveConnectorFactory in Velox.

Reply:
@majetideepak : That wasn't a great example from my side. 

As I think more about it I also realized that we still need some metadata service at the co-ordinator for reading and writing to work e2e as they all need DataSources/DataSinks to be built for them. So Hive + them is a good option. 

But if we have multiple metadata stores, then they should all be registered only once I feel. So separating them from connectors is still a good idea.

Sounds like you plan to do that in the future anyways. Maybe leave a comment about it.

Reply:
Well kind off. But I would separate them into their own functions. There is a registerFileSystems() function already.

Reply:
@aditi-pandit I moved the registrations to their own functions.